### PR TITLE
[exporter/clickhouseexporter]: persistent queue support

### PR DESCRIPTION
**Pull request by fredthomsen**
Addresses #27653.

**Description:**
Added persistent storage queue support by leveraging default `exporthelper.QueueSettings` config structure.  

**NOTE** This does end up being a **breaking** change to the API.

**Link to tracking Issue:** 
#27653 

**Testing:**
Adjust existing unit test.

### CODE REVIEW

File: .chloggen/clickhouseexporter-persistentqueue-support.yaml
Comment (songy23) [inline .chloggen/clickhouseexporter-persistentqueue-support.yaml:18]: Can you explain a bit here on how this becomes a breaking change?

**Review by songy23 — APPROVED** [review]
Minor comments, LGTM otherwise

File: .chloggen/clickhouseexporter-persistentqueue-support.yaml
Comment (fredthomsen) [reply .chloggen/clickhouseexporter-persistentqueue-support.yaml:18]: Thanks for pointing this out.  I had just taken the default config that the `exporterhelper.QueueSettings` provided, and thus the number of default consumers would have changed from 1 to 10.  However, that was a bit silly, so I've just kept the previous default in place.

File: exporter/clickhouseexporter/config.go
Comment (dmitryax) [inline exporter/clickhouseexporter/config.go:68]: This is not needed. `Validate()` is called for every embedded struct, see https://github.com/open-telemetry/opentelemetry-collector/blob/f0bba0ec3521fb726b17137d1c5672566aeb79f5/component/config.go#L50

File: exporter/clickhouseexporter/config.go
Comment (dmitryax) [inline exporter/clickhouseexporter/config.go:21]: Technically, this is a Go API breaking change, which is totally ok. We just need to add another changelog entry with `change_logs: []`

**Comment by dmitryax** [comment]
@hanjm, @Frapschen, PTAL. Would there be any problem if users configure multiple consumers? If so, we should document it.

File: exporter/clickhouseexporter/config.go
Comment (fredthomsen) [reply exporter/clickhouseexporter/config.go:68]: Ah thanks for this info.

File: .chloggen/clickhouseexporter-persistentqueue-support.yaml
Comment (dmitryax) [inline .chloggen/clickhouseexporter-persistentqueue-support.yaml:27]: This changelog entry should be kept for user only. We need another one marked `breaking` for `api` with wording like: "Replace `Config.QueueSettings` field with `exporterhelper.QueueSettings` and remove `QueueSettings` struct"

Comment (fredthomsen) [reply .chloggen/clickhouseexporter-persistentqueue-support.yaml]: Apologizes, was being a bit dense here.  Should be good now.

**Review by dmitryax — APPROVED** [review]
LGTM but I would like codeowners, @hanjm and @Frapschen, to take a look before merging

**Review by hanjm — APPROVED** [review]

**Comment by cwegener** [comment]
> @hanjm, @Frapschen, PTAL. Would there be any problem if users configure multiple consumers? If so, we should document it.

FWIW. I have been running the clickhouseexporter with a custom patch pretty much identical to the patch implemented here.

With one difference, I also removed the single consumer limit and ran with the default from the exporterhelper (10 consumers??), resulting in significant increase in overall pipeline throughput capacity.

Thanks @fredthomsen  for creating this PR!

**Pull request merged**
